sys/vm/vm_cheri_revoke.c

```diff
@@ -192,15 +199,14 @@ vm_cheri_revoke_kproc(void *arg __unused)
 		/*
 		 * Do the actual revocation pass.
 		 */
-		error = vm_cheri_revoke_pass_locked(&arc->cookie);
+		error = vm_cheri_revoke_pass_locked(arc->vm, &arc->cookie);
```

Comment:
It looks a bit weird to use this after it was seemingly free'd above.  It might be clearer to the reader with a comment and a local `vm` temporary like so:

```
     vmspace_switch_aio(&arc->vm);
     vm = arc->vm;

     /*
      * Drop arc's reference on the vmspace.  The scanner kproc holds a reference via
      * its p_vmspace pointer while the scanning the address space.  Dropping the
      * reference here permits the scan to return early if target process exits early
      * leaving the scanner kproc's reference as the only reference.
      */
     vmspace_free(arc->vm);
```